### PR TITLE
Reactify show page

### PR DESCRIPTION
**Pull request by jarmstro0**
Added react component to the show page. Added enzyme testing with a mock fetch and a controller test for the API.

### CODE REVIEW

**Comment by prophetsfall** [comment]
Please fix the failing tests and then resubmit

File: app/javascript/react/components/App.js [hunk 1]
```diff
@@ -1,7 +1,23 @@
 import React from 'react'
+import { Router, browserHistory, Route, IndexRoute } from 'react-router';
+
```
Comment (CassandraNewell) [inline app/javascript/react/components/App.js:3]: Watch all these extra lines!!!

File: app/javascript/react/components/Stop.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    fetch(`/api/v1/stops/${stop_id}`)
+      .then(response => {
+        if (response.ok) {
+          return response;
+        } else {
+          let errorMessage = `${response.status}(${response.statusText})` ,
+          error = new Error(errorMessage);
+          throw(error);
+        }
+        })
```
Comment (CassandraNewell) [inline app/javascript/react/components/Stop.js]: Watch your indentation too!

File: app/javascript/react/components/Stop.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          this.setState({stop: body})
+        })
+        .catch(error => console.error(`Error in fetch: ${error.message}`));
+  }
+
+  render(){
+    return(
+      <div>
+        <div className="id">
+          Id: {this.state.stop.id}
```
Comment (CassandraNewell) [inline app/javascript/react/components/Stop.js]: params should be lowercase (`id`)

File: spec/features/stops_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    it "returns http success" do
+      new_stop = Stop.create(name: "Porter")
+      get :show, params: {id: new_stop.id}
+      response.status.should eq(200)
+    end
+
+    it "returns the correct stop name" do
+      new_stop = Stop.create(name: "Porter")
+      get :show, params: {id: new_stop.id}
+      sleep(1)
```
Comment (CassandraNewell) [inline spec/features/stops_controller_spec.rb]: You shouldn't need this `sleep`, I don't think?

File: spec/features/user_visits_show_page_spec.rb [hunk 5]
```diff
@@ -1,17 +1,18 @@
 require 'rails_helper'
 require 'launchy'
 
-feature 'user visits show page'  do
-  scenario 'user sees the correct stop' do
+# feature 'user visits show page'  do
```
Comment (CassandraNewell) [inline spec/features/user_visits_show_page_spec.rb]: It is verboten to commit commented-out code! 🛑

File: spec/javascript/showStopTest.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    )
+  });
+
+  afterEach(() => fetchMock.restore)
+
+  it('should show labels, fetch data from api and present', (done) => {
+    setTimeout(() => {
+      expect(component.find('.id')).toBePresent()
+      expect(component.find('.name')).toBePresent()
+      expect(component.state('stop')).toEqual({ id:1, name:"El Greco",created_at:"2019-04-22T18:53:17.323Z",updated_at:"2019-04-22T18:53:17.323Z" });
```
Comment (CassandraNewell) [inline spec/javascript/showStopTest.js]: Please break into different lines

File: app/javascript/react/components/App.js [hunk 7]
```diff
@@ -1,7 +1,23 @@
...
+  // debugger
+  return (
+
+    <Router history={browserHistory}>
+      <Route path='/' component={Stop} >
+        // <IndexRoute component={Stop} />
+      </Route>
+      <Route path="stops/:id" component={Stop}/>
+    </Router>
+
```
Comment (CassandraNewell) [inline app/javascript/react/components/App.js]: Remove extra lines

**Review by CassandraNewell — CHANGES_REQUESTED** [review]

File: app/javascript/react/components/Stop.js (hunk 3 above)
Comment (jarmstro0) [reply app/javascript/react/components/Stop.js]: This is a label

File: app/javascript/react/components/App.js [hunk 8]
```diff
@@ -1,7 +1,18 @@
...
+import { Router, browserHistory, Route, IndexRoute } from 'react-router';
+
+import Stop from './Stop'
 
 export const App = (props) => {
-  return (<h1>Make It So React</h1>)
+  return (
+    <Router history={browserHistory}>
+      <Route path='/' component={Stop}>
+        //
```
Comment (CassandraNewell) [inline app/javascript/react/components/App.js]: Please delete this empty comment, too!

File: spec/javascript/showStopTest.js (hunk 6 above)
Comment (CassandraNewell) [reply spec/javascript/showStopTest.js]: for long objects, break into multiple lines:
```suggestion
      expect(component.state('stop')).toEqual({ 
        id:1, 
        name:"El Greco",
        created_at:"2019-04-22T18:53:17.323Z",
        updated_at:"2019-04-22T18:53:17.323Z"
      });
```

**Review by CassandraNewell — CHANGES_REQUESTED** [review]

File: app/controllers/stops_controller.rb [hunk 9]
```diff
@@ -7,4 +7,20 @@ def show
...
 
+  def create
+    new_stop = Stop.new(params[:stop])
+
+    if new_stop.save
+      flash[:notice] = "stop created successfully"
+
+    else
+      flash[:error] = "error"
+
```
Comment (CassandraNewell) [inline app/controllers/stops_controller.rb:18]: 😭you guys are killing me with your extra lines, but I'll let it go!

**Review by CassandraNewell — APPROVED** [review]

File: app/controllers/stops_controller.rb (hunk 9 above)
Comment (jarmstro0) [reply app/controllers/stops_controller.rb:18]: sorry about that. that file accidentally got included when we switched branches. I'll be sure to throw in a few extra lines for you next time though :)

**Pull request merged**